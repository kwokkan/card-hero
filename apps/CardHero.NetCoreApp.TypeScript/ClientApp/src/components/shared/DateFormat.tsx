import React from "react";
import { relativeFromNow, toISOString } from '../../utils/date';

interface IDateFormatProps {
    date: Date;
}

export function DateFormat(props: IDateFormatProps) {
    const d = relativeFromNow(props.date);

    return (
        <time title={props.date.toUTCString()} dateTime={toISOString(props.date)}>{d}</time>
    );
}
