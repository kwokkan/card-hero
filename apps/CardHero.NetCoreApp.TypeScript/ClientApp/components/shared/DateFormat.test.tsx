import MockDate from "mockdate";
import React from "react";
import renderer from "react-test-renderer";
import { DateFormat } from "./DateFormat";

const mockDate = new Date(2019, 0, 2, 3, 4, 5, 6);

beforeEach(() => {
    MockDate.set(mockDate);
});

afterEach(() => {
    MockDate.reset();
});

test("<DateFormat /> renders correctly", () => {
    const tree = renderer
        .create(<DateFormat date={mockDate} />)
        .toJSON();

    expect(tree).toMatchSnapshot();
});
