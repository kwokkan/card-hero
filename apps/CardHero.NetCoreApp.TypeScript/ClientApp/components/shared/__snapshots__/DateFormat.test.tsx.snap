// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateFormat /> renders correctly 1`] = `
<time
  dateTime="2019-01-02T03:04:05.006Z"
  title="Wed, 02 Jan 2019 03:04:05 GMT"
>
  a few seconds ago
</time>
`;
